File: Scala/AdventOfCodeDat1.scala
/**
  * Created by domin_000 on 23.01.2016.
  */
object Main {
  def main(args: Array[String]) {
    val res  = floor("(()(()(")
    println(res)
  }
  def floor(text:String): Int ={
    val up = text.toCharArray.count(x => x == '(')
    val down = text.toCharArray.count(x => x == ')')
    up - down
  }
}

File: Elixir/AdventOfCodeDat1.ex
defmodule Utils do
  def floor(text) do
    up |> Enum.map(fn x -> String.split(x, "") end) |> Enum.count(fn x -> x == "(")
    down |> Enum.map(fn x -> String.split(x, "") end) |> Enum.count(fn x -> x == ")")
    up - down
  end
end

IO.inspect Utils.floor("(()(()(")
